nixos/modules/services/networking/headscale.nix

```diff
@@ -11,7 +11,14 @@ with lib; let
   runDir = "/run/headscale";
 
   settingsFormat = pkgs.formats.yaml {};
-  configFile = settingsFormat.generate "headscale.yaml" cfg.settings;
+  settingsRewritten = cfg.settings // lib.optionalAttrs (cfg.settings.oidc.client_secret_file != null) {
+    # Overriding the path to the secret file, because headscale should see the
+    # where systemd places its secret, not the original path. This allows the
+    # secret to be readable only by root, and systemd managing its access. The
+    # secret is loaded by systemd's LoadCredential.
+    oidc.client_secret_file = "\${CREDENTIALS_DIRECTORY}/hs_client_oidc_secret";
```

Comment:
And here you tell headscale to not use the secretfile that we already knew about, but rather use the one from systemd?

Or am I missing something? does user have to add the secrets into systemd cred store in a different way?

Reply:
Yes. The original idea was to:
- the original credential path goes to systemd.
- the "derived" credential path goes to headscale.

I now think there is a better way, commented above in https://github.com/NixOS/nixpkgs/pull/217482#discussion_r1126503743 .